src/NHibernate/Dialect/Function/CastFunction.cs

```diff
@@ -87,23 +87,29 @@ public SqlString Render(IList args, ISessionFactoryImplementor factory)
 				throw new QueryException(string.Format("invalid Hibernate type for cast(): type {0} not found", typeName));
 			}
 
-			if (CastingIsRequired(sqlType))
-			{
-				return new SqlString("cast(", args[0], " as ", sqlType, ")");
-			}
-			else
-			{
+			//TODO 6.0: Remove pragma block with its content
+#pragma warning disable 618
+			if (!CastingIsRequired(sqlType))
 				return new SqlString("(", args[0], ")");
-			}
+#pragma warning restore 618
+
+			return Render(args, sqlType, factory);
 		}
 
 		#endregion
 
+		// Since v5.3
+		[Obsolete("This method has no usages and will be removed in a future version")]
 		protected virtual bool CastingIsRequired(string sqlType)
 		{
 			return true;
 		}
 
+		protected virtual SqlString Render(IList args, string sqlType, ISessionFactoryImplementor factory)
```

Comment:
We should xml document members meant to be used or overridden by user code.

```suggestion
		/// <summary>
		/// Renders the SQL fragment representing the SQL cast.
		/// </summary>
		/// <param name="args">The cast arguments.</param>
		/// <param name="sqlType">The SQL type to cast to.</param>
		/// <param name="factory">The session factory.</param>
		/// <returns>A SQL fragment.</returns>
		protected virtual SqlString Render(IList args, string sqlType, ISessionFactoryImplementor factory)
```

I wonder if we should replace `args` with just an `expression` parameter with `args[0]` into it. It is less versatile, in case someone would want to recompute differently the `sqlType`. But for this need I say we should instead make the main `Render` virtual too. While here we have a kind of redundancy between `args` and `sqlType`, causing `args[1]` to be useless in all current cases.

Reply:
> I wonder if we should replace args with just an expression parameter with args[0] into it.

I think we should. Replaced `args` with `expression` and made the main `Render` method virtual.